Can't compile with gcc 4.*

Code from ImathConfig.h(generated from ImathConfig.h.in) fails to compile with gcc 4.*(tried 4.8 and 4.9). on 5.* it compiles fine:
```
#define IMATH_ENABLE_API_VISIBILITY

// MSVC does not do the same visibility attributes, and when we are
// compiling a static library we won't be in DLL mode, but just don't
// define these and the export headers will work out
#if ! defined(_MSC_VER) && defined(IMATH_ENABLE_API_VISIBILITY)
#  define IMATH_PUBLIC_SYMBOL_ATTRIBUTE __attribute__ ((__visibility__ ("default")))
#  define IMATH_PRIVATE_SYMBOL_ATTRIBUTE __attribute__ ((__visibility__ ("hidden")))
// clang differs from gcc and has type visibility which is needed for enums and templates
#  if __has_attribute(__type_visibility__)
#    define IMATH_PUBLIC_TYPE_VISIBILITY_ATTRIBUTE __attribute__ ((__type_visibility__ ("default")))
#  endif
#endif
```
Is it possible to support gcc 4.*?

It looks like preprocessor parsing error, not related to c++ standard
```
OpenEXRConfig.h:100:22: error: missing binary operator before token "("
 #  if __has_attribute(__type_visibility__)
                              ^
gmake[2]: *** [src/lib/Iex/CMakeFiles/Iex.dir/IexBaseExc.cpp.o] Error 1
gmake[1]: *** [src/lib/Iex/CMakeFiles/Iex.dir/all] Error 2
gmake: *** [all] Error 2
```

--- a/config/ImathConfig.h.in
+++ b/config/ImathConfig.h.in
@@ -130,6 +130,13 @@
 #endif
 
 
+// On modern versions of gcc & clang, __has_attribute can test support for
+// __attribute__((attr)).  Make sure it's safe for other compilers.
+#ifndef __has_attribute
+#    define __has_attribute(x) 0
+#endif
+
+
 //
 // Simple way to mark things as deprecated.
 // When we are sure that C++14 is our true minimum, then we can just
@@ -156,7 +163,7 @@
 #  define IMATH_PUBLIC_SYMBOL_ATTRIBUTE __attribute__ ((__visibility__ ("default")))
 #  define IMATH_PRIVATE_SYMBOL_ATTRIBUTE __attribute__ ((__visibility__ ("hidden")))
 // clang differs from gcc and has type visibility which is needed for enums and templates
-#  if defined(__has_attribute) && __has_attribute(__type_visibility__)
+#  if __has_attribute(__type_visibility__)
 #    define IMATH_PUBLIC_TYPE_VISIBILITY_ATTRIBUTE __attribute__ ((__type_visibility__ ("default")))
 #  endif
 #endif

--- a/config/ImathConfig.h.in
+++ b/config/ImathConfig.h.in
@@ -156,7 +156,7 @@
 #  define IMATH_PUBLIC_SYMBOL_ATTRIBUTE __attribute__ ((__visibility__ ("default")))
 #  define IMATH_PRIVATE_SYMBOL_ATTRIBUTE __attribute__ ((__visibility__ ("hidden")))
 // clang differs from gcc and has type visibility which is needed for enums and templates
-#  if __has_attribute(__type_visibility__)
+#  if defined(__has_attribute) && __has_attribute(__type_visibility__)
 #    define IMATH_PUBLIC_TYPE_VISIBILITY_ATTRIBUTE __attribute__ ((__type_visibility__ ("default")))
 #  endif
 #endif

